src/manifest/index.ts

```diff
@@ -39,6 +39,13 @@ import {
   IHDRInformation,
 } from "./types";
 
+interface IContentContext {
```

Comment:
Why declaring this structure here instead of directly in `get_needed_segment.ts`?
The problem I see is that there is technically much more plausible combinations (you might want to add a `segment` or removes the `representation` property).

Defining them all here is going to lead to naming issues1

Reply:
That point still stands for me